orangecontrib/spectroscopy/widgets/owspectra.py

```diff
@@ -1009,6 +1067,71 @@ def grid_apply(self):
         self.plot.showGrid(self.show_grid, self.show_grid, alpha=0.3)
         self.show_grid_a.setChecked(self.show_grid)
 
+    def find_peak_variables(self):
+        if self.maximum_point is None:
+            self.maximum_point = np.amax(self.data.X[:, self.data_xsind])
```

Comment:
amax is problematic for data with unknown values (nans)

Reply:
Also, ordering is not needed here.

Reply:
Also, does it really have to be computed from the whole data set or just for the data shown? I'd say only on the data shown because this is much faster. And then these can be recomputed when the curves that are shown are resampled.